src/osdc/Objecter.h

```diff
@@ -1645,6 +1645,10 @@ class Objecter : public md_config_obs_t, public Dispatcher {
   std::atomic<bool> initialized{false};
 
 private:
+  struct fastfail {
+    std::unordered_set<pg_t> pgs;
+    std::shared_mutex lock;
```

Comment:
I'm not sure that I'm seeing everything I'm expecting with a reader-writer lock;  which paths take it read vs write, can you point to them in the code?

Reply:
looks like this can be a plain mutex